### PR TITLE
MAINT: Get docs to build again

### LINKED ISSUE #380: Docs don't build
**Problem Description**

Not sure about the underlying issue (or why this has cropped up now) but doing a simple `make html` in the docs folder results in a traceback in one of the Sphinx extensions.

```
05:10 $ cat /var/folders/yx/p6tl77d502qc2v1srt84n91r0000gq/T/sphinx-err-a9YfOQ.log
# Sphinx version: 1.5.5
# Python version: 2.7.13 (CPython)
# Docutils version: 0.13.1 release
# Jinja2 version: 2.9.6
# Last messages:

# Loaded extensions:
Traceback (most recent call last):
  File "/Users/jvankerschaver/.edm/envs/docs-generating-env/lib/python2.7/site-packages/sphinx/cmdline.py", line 295, in main
    opts.warningiserror, opts.tags, opts.verbosity, opts.jobs)
  File "/Users/jvankerschaver/.edm/envs/docs-generating-env/lib/python2.7/site-packages/sphinx/application.py", line 199, in __init__
    self.setup_extension(extension)
  File "/Users/jvankerschaver/.edm/envs/docs-generating-env/lib/python2.7/site-packages/sphinx/application.py", line 512, in setup_extension
    mod = __import__(extension, None, None, ['setup'])
  File "/Users/jvankerschaver/src/enthought/chaco/docs/source/sphinxext/traitsdoc.py", line 7, in <module>
    from . import docscrape
ValueError: Attempted relative import in non-package
```

**Reproduction Steps:**

- Install the package
- `cd docs; make html`

### PR DESCRIPTION
**Pull request by jvkersch**
To get the docs build back on the rails (#380), I reverted the Python 2/3 compatibility layer from https://github.com/enthought/chaco/pull/369 for the Sphinx extensions (the rest of the compatibility layer is still in place). I began by updating the import statements to match what Sphinx expects, but even after doing this I found that some of the six constructs didn't play well with the rest of the code. As this code is old and will not work on Python 3 without some serious maintenance (#324), I decided to just roll back that part of the code.

closes #380

### CODE REVIEW

File: docs/source/sphinxext/docscrape.py [hunk 1]
```diff
@@ -441,8 +437,8 @@ def __str__(self):
                  'meth': 'method'}
 
         if self._role:
-            if self._role not in roles:
-                print("Warning: invalid role %s" % self._role)
+            if not roles.has_key(self._role):
```
Comment (corranwebster) [inline docs/source/sphinxext/docscrape.py]: This seems an unnecessary change.

File: docs/source/sphinxext/docscrape.py [hunk 2]
```diff
@@ -419,7 +415,7 @@ def __init__(self, func, role='func'):
                 argspec = inspect.formatargspec(*argspec)
                 argspec = argspec.replace('*','\*')
                 signature = '%s%s' % (func_name, argspec)
-            except TypeError as e:
+            except TypeError, e:
```
Comment (corranwebster) [inline docs/source/sphinxext/docscrape.py]: Don't revert this line!

File: docs/source/sphinxext/docscrape.py [hunk 3]
```diff
@@ -403,13 +399,13 @@ def __init__(self, func, role='func'):
...
         try:
             NumpyDocString.__init__(self,inspect.getdoc(func) or '')
-        except ValueError as e:
-            print('*'*78)
-            print("ERROR: '%s' while parsing `%s`" % (e, self._f))
-            print('*'*78)
-            #print("Docstring follows:")
-            #print(doclines)
-            #print('='*78)
+        except ValueError, e:
```
Comment (corranwebster) [inline docs/source/sphinxext/docscrape.py]: Or this line!

File: docs/source/sphinxext/numpydoc.py [hunk 4]
```diff
@@ -282,7 +275,7 @@ def _import_by_name(name):
             return obj
         else:
             return sys.modules[modname]
-    except (ValueError, ImportError, AttributeError, KeyError) as e:
+    except (ValueError, ImportError, AttributeError, KeyError), e:
```
Comment (corranwebster) [inline docs/source/sphinxext/numpydoc.py]: Again, keep the 'as'

**Comment by jvkersch** [comment]
Thanks Corran. Just for the record, the reverts were done completely mechanistically. I'd rather not touch this part of the codebase for the time being; I agree that using standard Sphinx plugins like Napoleon and the Traits documenter (which were probably not around when this code was originally written) is the way forward.
